docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,296 @@
-# Manifest factorization through `.flux.yaml` configuration files
+# Manifest generation through `.flux.yaml` configuration files
 
-## Enabling search of `.flux.yaml` files
+This feature lets you generate Kubernetes manifests with a program,
+instead of having to include them in your git repo as YAML files. For
+example, you can use `kustomize` to patch a common set of resources to
+suit a particular environment.
 
-To enable it supply the command-line flag `--manifest-generation=true` to `fluxd`.
+Manifest generation is controlled by the flags given to `fluxd`, and
+`.flux.yaml` files in your git repo.
 
-## Goal
+To enable it, you will need to
 
-It is a common pattern to run very similar resources in separate clusters. There are various scenarios in which this is
-required, the two main ones being:
+ 1. pass the command-line flag `--manifest-generation=true`
+to `fluxd`.
+ 2. put at least one `.flux.yaml` file in the git repository.
 
-* Having a staging/canary cluster and production cluster. The resources from the staging cluster are regularly promoted
-  to the production cluster. In addition, there are long-term differences between the cluster resources (e.g. different
-  security keys, different database endpoints etc ...).
-* Federation. Different clusters run in separate regions, usually with very similar resources but different
-  configurations.
+Where to put `.flux.yaml`, and what should be in it, are described in
+the sections following.
 
-The main goal of `.flux.yaml` configuration files is to help deploying similar resources/clusters:
+## Setting manifest generation up
 
-* with minimal replication of resource definitions
-* while keeping Flux neutral about the factorization technology used
+The command-line flag `--git-path` (which can be given multiple
+values) marks a "target path" within the git repository in which to
+find manifests. If `--git-path` is not supplied, the top of the git
```

Comment:
Maybe it's worth indicating that multiple `--git-paths` can be provided?

Reply:
You mean like saying "which can be given multiple values"?

Reply:
Yes. Note that it's also valid to provide multiple `--git-path` arguments (pflag will coalesce them)

Reply:
Yup, that's what I mean by "provide multiple values" -- use the flag to provide multiple values. I don't want to get into how the flag parsing works in this doc though.